frontend/hub/api/request.ts

```diff
@@ -0,0 +1,134 @@
+import { AnsibleError } from '../../common/crud/ansible-error';
+import { getCookie } from '../../common/crud/cookie';
+import { Delay } from '../../common/crud/delay';
+import { TaskResponse } from '../tasks/Task';
+
+interface Options {
+  method: string;
+  headers?: HeadersInit;
+  body?: BodyInit;
+  signal?: AbortSignal;
+}
+
+interface Response<T> {
+  statusCode: number;
+  response: T;
+}
+
+export async function request<T extends object | TaskResponse>(
```

Comment:
These seem like generic function names that are in a hub/api file.
Maybe they should be hub specific names like hubRequest.
How do there differ from the common functions like postRequest that is the in common folder?

Reply:
I will rename them to reflect that it is specific to the HUB project. The `request` function for the HUB deals with the 202 as a response, and either return the task response or the generic one passe to the function.